stdlib/list.ml

```diff
@@ -514,3 +535,17 @@ let of_seq seq =
       | Seq.Cons (x, next) -> x :: direct (depth-1) next
   in
   direct 500 seq
+
+module Syntax = struct
+
+  let return x = [x]
+
+  let ( let* ) x f = concat_map f x
+
+  let ( and* ) a b = product a b
```

Comment:
Why not `let ( and* ) = ( and+ )`? For that matter, why not `let ( and+ ) = product` since parameter order doesn't matter?